app/controllers/v3/auth/federation_controller.rb

```diff
@@ -1,9 +1,61 @@
 module V3
   module Auth
     class FederationController < ApplicationController
-      def oidc; end
+      TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%L000Z'.freeze
 
-      def voms; end
+      def oidc
+        credentials = ::Auth::Oidc.unified_credentials(ENV.select { |name| name.start_with?('OIDC') })
+        headers['X-Subject-Token'] = ::Tokenator.to_token(credentials.to_hash)
+        render json: response_hash(credentials, 'oidc')
+      end
+
+      def voms
+        credentials = ::Auth::Voms.unified_credentials(ENV.select { |name| name.start_with?('GRST', 'SSL') })
+        headers['X-Subject-Token'] = ::Tokenator.to_token(credentials.to_hash)
+        render json: response_hash(credentials, 'voms')
```

Comment:
I believe it is. However, it shouldn't matter here since @dudoslav is specifying the format explicitly via `:json`. I'm just not sure how this will work with `Accept` and `Content-Type` headers. For example, what happens when a client requests a different format? Or which formats will be offered by the server as supported formats if the client asks?